src/tests/Interop/PInvoke/Generics/GenericsNative.Vector256U.cpp

```diff
@@ -8,9 +8,15 @@
 
 #if defined(TARGET_XARCH)
     #include <immintrin.h>
+#elif defined(TARGET_ARMARCH)
+    // Intentionally empty
+#else
+    #error Unsupported target architecture
+#endif
 
+#if defined(__AVX2__)
```

Comment:
Which compilers define this and under what scenarios is it defined?

AFAIK, most compilers set things like `__AVX2__` based on what your compiler switches are, not based on what the underlying hardware supports. This would require us compiling these assemblies for `-march=native` on GCC/Clang and doing some CMake magic for `msvc`.

Reply:
You are right. I do not account for that. 
I found only one way perform CMake magic for arch detection, and I do not like that. This is it https://github.com/openMVG/openMVG/blob/f5ecb48a75f5ea2126930f559e6920fe2251baf9/src/cmakeFindModules/OptimizeForArchitecture.cmake#L357
This appears to be overkill for this only case where AVX detection needed.
Being a bit selfish, what if just add to this tests CMakeLists.txt
```
if (MSVC)
    # MSVC require proper detection of the host architecture, for now assume that AVX2 always used.
    add_compile_options(/arch:AVX2)
else()
    add_compile_options(-march=native)
endif()
```
In my understanding this improve situation with GCC and Clang, at the same time, preserve status quo in MSVC.
I know that this is just kicking can down the road, but maybe that can be acceptable compromise, since this is seems to be affect only minority of users and all of them so far was on Linux.

Reply:
That seems fine to me. I believe it might not be an issue on MSVC due to how the compiler handles `__m256` vs how Clang/GCC handle it.

The actual logic executing should be entirely dependent on the managed side `if (Avx.IsSupported)` check and so shouldn't execute.

Reply:
@tannergooding I enable `--march=native` only on Clang and GCC on non-ARM builds. Would this works in your opinion?

Reply:
Provided the code is actually compiled and works as expected on MSVC, then it looks good to me.

I'm not familiar enough with what each compiler does by default to know off the top of my head.

Reply:
@kant2002 Have you been able to verify the behavior here?

Reply:
@AaronRobinsonMSFT Now yes.